src/parser.h

```diff
@@ -0,0 +1,147 @@
+#pragma once
+
+#include "datatypes.h"
+
+enum { MAX_YAML_LINE_LENGTH = 64 };
+enum { MAX_YAML_NUM_LINES = 1024 };
+enum { NUM_YAML_FILE_SECTIONS = 3 };  // rules, image location, tiles
+enum {
+  NUM_RULES = 2
+};  // allow 90 degree rotations, allow 180 degree
+    // rotations
+
+/**
+ * Get the text from the input yaml file
+ *
+ * Given the pointer to the user submitted yaml file, generate an array of
+ * arrays of strings containing the contents of the yaml file (as plaintext).
+ * The zeroth array of strings contains the "rules" section, the first the
+ * "image location" section, the second the "tiles" section.
+ *
+ * @param input_yaml_file The pointer to the file object representing the input
+ * yaml
+ *
+ * @return The array of arrays of strings containing the contents of the file
+ * split into sections
+ */
+char*** get_text_split_sections(FILE* input_yaml_file);
+
+/**
+ * Parses the "rules" section of the input text.
+ *
+ * Given the pointer to the rules text, determine the status of the rules
+ * (currently if 90 and 180 degree rotations are allowed). Return an array of
+ * ints corresponding to the rule statuses.
+ *
+ * The values in the return array correspond to:
+ * [0]: 90 degree rotations
+ * [1]: 180 degree rotations
+ *
+ * @param rules_text The block of text containing the rules input, no trailing
+ * empty lines after it.
+ *
+ * @return An array of ints corresponding to rules statuses
+ */
+int* parse_rules(char** rules_text);
+
+/**
+ * Parses the "image location" section of the input text.
+ *
+ * Given the pointer to the image location text, determine the location of the
+ * images. Return a string corresponding to that location, with a trailing
+ * forward slash.
+ *
+ * @param image_location_text The section of the text containing the image
+ * location text, no trailing lines after it.
+ *
+ * @return The string corresponding to the location of the images with a
+ * trailing forward slash.
+ */
+char* parse_imdir(char** image_location_text);
```

Comment:
may want to define/specify functionality upon failure (ie no image dir found)

Reply:
Agreed, added description of failure behavior to docstring for this function and others. Will implement failure behavior